### PR TITLE
Fix the ACK Timer responder so that it passes the RDM tests

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
blocked behind #917, closes #854

**Comment by peternewman** [comment]
Can you resync please @nomis52

File: common/rdm/AckTimerResponder.cpp [hunk 1]
```diff
@@ -208,13 +208,29 @@ RDMResponse *AckTimerResponder::ResponseFromQueuedMessage(
...
+        request->SourceUID(),
+        request->TransactionNumber(),
+        RDM_ACK,
+        QueuedMessageCount(),
+        ROOT_RDM_DEVICE,
+        queued_response->Pid(),
+        queued_response->ParamData(),
+        queued_response->ParamDataSize());
     case RDMCommand::SET_COMMAND_RESPONSE:
-      return GetResponseFromData(request,
```
Comment (peternewman) [inline common/rdm/AckTimerResponder.cpp]: Ah, so it was just that it had the wrong PID in the response? It looks like GetResponseWithPid will be a neater fit, or does it have to come from the root device? Of course in that case you could extend GetResponseWithPid to have a version that passes in the subdevice too.

Comment (nomis52) [inline common/rdm/AckTimerResponder.cpp]: GetResponseWithPid doesn't work because you need to override the command class as well.

Comment (peternewman) [inline common/rdm/AckTimerResponder.cpp]: Ah yes, GET_COMMAND_RESPONSE becomes GET_RESPONSE.

**Comment by peternewman** [comment]
LGTM

File: common/rdm/AckTimerResponder.cpp (hunk 1 above)
Comment (nomis52) [inline common/rdm/AckTimerResponder.cpp]: Not just that, a GET_REQUEST can reply with a SET_RESPONSE. Queued messages are funky.

Comment (peternewman) [inline common/rdm/AckTimerResponder.cpp]: Ah, I'd missed the fact you were switching on the queued message!

**Pull request merged**